content/learn/book/getting-started/setup/_index.md

```diff
@@ -85,34 +85,39 @@ bevy = "0.5" # make sure this is the latest version
 ```
 ### Enable Fast Compiles (Optional)
 
-Bevy can be built just fine using default configuration on stable Rust. However for maximally fast iterative compiles, we recommend the following configuration:
+Bevy can be built just fine using default configuration on stable Rust. However there are a few things you can do to significantly improve iterative compile times:
 
-* **Enable Bevy's Dynamic Linking Feature**: This is the biggest iterative compilation time win and requires no special setup. When you have `bevy` as a dependency you can run your app with dynamic linking like this:
+If you're using Windows, Bevy's dynamic linking feature won't work unless you've followed the three extra steps below first.
+
+* **Bevy's Dynamic Linking Feature**: This is the biggest iterative compilation time win and requires no special setup unless you use Windows. When you have `bevy` as a dependency you can run your app with dynamic linking like this:
```

Comment:
Does this actually require special setup on windows? Shouldn't it work with the default setup? If this is just a reference to the "share generics" issue, we should just cover that in the "fast compiles configuration section", either by removing it like you did in your pr or by documenting the weirdness / potential errors in the fast compiles section.

I'd prefer it if the "dynamic linking" section was "simple and asterisk free"

Reply:
tl;dr: My understanding is that the problem is the share-generics issue. It seems like share-generics are on by default in debug compilation, so that's why it doesn't work on Windows without special setup.

---

I did some more testing to make sure if special setup is required, and in my tests I run into some variation of the errors linked in https://github.com/bevyengine/bevy-website/issues/131 when running with `--features bevy/dynamic` on Windows unless I run with..

**Option 1**
- `rustflags = ["-Zshare-generics=n"]`
- Nightly (required because of the -Z flag in rustflags)

**Option 2**
- `--release`

Like you write, `share-generics` *is* off by default, but only with `--release`, otherwise it's apparently on by default. This is where I got this information https://github.com/rust-lang/rust/issues/67276 and I've verified it on my machine by successfully doing `cargo +stable run --features bevy/dynamic --release` with no config.toml file. Incremental compiles with `--features bevy/dynamic --release` is slower for me than building in debug without bevy/dynamic.

So, following option 1, with nightly and the config file you've already followed 2 of the 3 "advanced" points, so you might as well do the remaining step of using LLD as well. If nothing else, just to keep the instructions simple.

I also want to keep these instructions as simple as possible, and it sucks to have the extra info for Windows users. But it seems to me like bevy/dynamic simply does not work on Windows unless you at least compile on nightly and with `rustflags = ["-Zshare-generics=n"]`.

Reply:
Thanks for the breakdown! I'll run a few tests on my machine just to get a feel for the various permutations / see if these behaviors are reproducible.

It does sound like we'll probably need to roll with the approach in this pr  :smile: